tests/integration/cache_test.go

```diff
@@ -144,6 +252,73 @@ func testWatch(t *testing.T, kv clientv3.KV, watcher Watcher) {
 	}
 }
 
+func TestCacheRejectsInvalidWatchOptions(t *testing.T) {
```

Comment:
This test should be merged with `TestCacheRejectsUnsupportedWatch`. We should have single test to track what options are supported/work.